cms/djangoapps/contentstore/outlines.py

```diff
@@ -264,50 +334,12 @@ def _make_section_data(section, unique_sequences):
             continue
         else:
             unique_sequences[sequence.location] = section
-        seq_user_partition_groups, error = _make_user_partition_groups(
-            sequence.location, sequence.group_access
-        )
-        if error:
-            section_errors.append(error)
 
-        # Bubble up User Partition Group settings from Units if appropriate.
-        sequence_upg_from_units = _bubbled_up_groups_from_units(
-            [unit.group_access for unit in sequence.get_children()]
-        )
-        for user_partition_id, group_ids in sequence_upg_from_units.items():
-            # If there's an existing user partition ID set at the sequence
-            # level, we respect it, even if it seems nonsensical. The hack of
-            # bubbling things up from the Unit level is only done if there's
-            # no conflicting value set at the Sequence level.
-            if user_partition_id not in seq_user_partition_groups:
-                section_errors.append(
-                    _make_bubbled_up_error(sequence.location, user_partition_id, group_ids)
-                )
-                seq_user_partition_groups[user_partition_id] = group_ids
-            else:
-                section_errors.append(
-                    _make_not_bubbled_up_error(
-                        sequence.location, sequence.group_access, user_partition_id, group_ids
-                    )
-                )
+        sequence_data, sequence_errors = _make_sequence_data(sequence)
```

Comment:
BTW, good call on extracting this out.

Reply:
I think the easiest way to pipe this data through without writing back to the XBlock is go make it so that there's a dict that holds that extra user partition group mapping data for sequences that are children of a split_test. You can add stuff to that dict in your calls to `_make_split_test_data` earlier, and then pass values from that dict as an extra param to `_make_sequence_data`, and then let `_make_sequence_data` stitch it together.